### PR TITLE
Checksummed address

### PR DESCRIPTION
**Pull request by dakingha69**
Solves #23

### CODE REVIEW

File: src/CurrencyNetwork.ts [hunk 1]
```diff
@@ -10,21 +11,21 @@ export class CurrencyNetwork {
   }
 
   public getInfo (networkAddress: string): Promise<any> {
-    if (!this.utils.checkAddress(networkAddress)) {
+    if (!ethUtils.isValidAddress(networkAddress)) {
```
Comment (berndbohmeier) [inline src/CurrencyNetwork.ts]: Why do you not check for a checksum address here?

**Comment by berndbohmeier** [comment]
And I think the clientlib should expose a function that checks if an address is valid. This can then be used by the mobileapp for validating input

File: src/CurrencyNetwork.ts (hunk 1 above)
Comment (dakingha69) [reply src/CurrencyNetwork.ts]: didnt want to make the check to strict originally.. but yea we should always check for checksum at this point

Comment (berndbohmeier) [reply src/CurrencyNetwork.ts]: Yes maybe it is to strict. Can you make it similar to this https://github.com/ethereum/eth-utils/blob/master/eth_utils/address.py#L80
So only check for checksum address if there are mixed case letter? if all are lower case or all are upper case it was probably not intended to get checked for a checksum.

File: tests/e2e/Trustline.test.ts [hunk 2]
```diff
@@ -117,8 +117,8 @@ describe('e2e', () => {
             setTimeout(() => {
               tl2.trustline.getUpdates(networkAddress).then(updates => {
                 const latestUpdate = updates[updates.length - 1]
-                expect(latestUpdate.address).to.equal(user1.address)
-                expect(latestUpdate.amount).to.equal(1500)
+                expect(latestUpdate.address.toLowerCase()).to.equal(user1.address.toLowerCase())
```
Comment (berndbohmeier) [inline tests/e2e/Trustline.test.ts]: Did that slip in by accident? There should be no lowercase needed anymore right?

File: src/CurrencyNetwork.ts [hunk 3]
```diff
@@ -10,21 +11,21 @@ export class CurrencyNetwork {
   }
 
   public getInfo (networkAddress: string): Promise<any> {
-    if (!this.utils.checkAddress(networkAddress)) {
+    if (!ethUtils.isValidChecksumAddress(networkAddress)) {
```
Comment (berndbohmeier) [inline src/CurrencyNetwork.ts]: And please use the new created checkAddress function also here in the code, so it is the same everywhere

**Pull request merged**